src/components/history/InfiniteList.js

```diff
@@ -106,25 +113,32 @@ export default class InfiniteList extends PureComponent {
       if (width !== 0) {
         this.cache.clearAll()
         this.list.recomputeRowHeights()
-        this.debounedScrollToRowBeforeResize()
+        // if this.didRenderLastRow is active during the scrollLocked phase
+        // this.onRowsRendered will make sure to always stick to the bottom
+        // and we won't show the recalculate overlay
+        //
+        // this stick to the bottom case is the most common one and that's
+        // why we decided to make this optimization for it
+        if (this.didRenderLastRow) {
+          this.debouncedReleaseLockState()
+        } else {
+          this.debouncedScrollToRowBeforeResize()
+        }
       }
     }
     this.prevWidth = width
   }
 
-  debounedScrollToRowBeforeResize = debounce(() => {
-    // When the last row was rendered the user was either at the bottom or very close
-    // to the bottom of the chat. Jumping to the last row was the best experience in
-    // this case.
-    if (this.didRenderLastRow) {
-      this.scrollToRow(this.props.rows.length - 1)
-    } else {
-      const newIndex = findIndex(
-        this.props.rows,
-        item => item.id === this.idOfMessageToBeFocusedAfterResize,
-      )
-      this.list.scrollToRow(newIndex)
-    }
+  debouncedScrollToRowBeforeResize = debounce(() => {
+    const newIndex = findIndex(
+      this.props.rows,
+      item => item.id === this.idOfMessageToBeFocusedAfterResize,
+    )
+    this.list.scrollToRow(newIndex)
+    this.setState({ scrollLocked: false })
+  }, 700)
```

Comment:
Maybe we should decrease this value? Because for the unread counter label we have 100ms debounce to mark a current channel as read.

Reply:
good point, the issue here though is that we can't go much lower. We can do 400ms for example, but with lags react-virtualized is extremely slow and can take quite a while to render and re-calculate everything

Reply:
ok, got it. Yes, especially in IE:)